### PR TITLE
Remove React.FC from all components

### PR DESCRIPTION
**Pull request by liby**
## Motivation

[`React.FC` is unnecessary: it provides next to no benefits and has a few downsides. (See below.) I see a lot of beginners to TS+React using it, however, and I think that it's usage in this template is a contributing factor to that, as the prominence of this template makes it a de facto source of "best practice".](https://github.com/facebook/create-react-app/pull/8177)

## Changes

Using [jscodeshift](https://github.com/facebook/jscodeshift) and [codemod-replace-react-fc-typescript](https://github.com/gndelia/codemod-replace-react-fc-typescript) to remove `React.FC` from components.

### CODE REVIEW

File: tools/executors/figmaRes/figma/generateReactIcon.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ module.exports = async function generateReactIcon(name, svgCode, customStyles) {
     const colors = getColors(colorIdx);
 
     return `
-import { FC } from 'react';
+
```
Comment (lawvs) [inline tools/executors/figmaRes/figma/generateReactIcon.js:103]: We use scripts to generate the icons automatically, you may need to re-run the script to update them.

```suggestion
```

See https://github.com/toeverything/AFFiNE/blob/e3e37413930eb0a52f1b030f2d04d049152d003f/docs/how-to-auto-download-figma-assets-in-affine.md

File: tools/executors/svgOptimize/svgo.js [hunk 2]
```diff
@@ -88,9 +88,9 @@ async function getJSXContent(name, svgCode) {
     }
     let matcher = svgrContent.match(/<svg ([^\>]+)>([\s\S]*?)<\/svg>/);
     return `
-import { FC } from 'react';
+
```
Comment (lawvs) [inline tools/executors/svgOptimize/svgo.js:91]:
```suggestion
```

**Review by lawvs — COMMENTED** [review]
Thanks for your PR! There are very minor nit.

File: apps/ligo-virgo/src/pages/tools/icons/Icons.tsx [hunk 3]
```diff
@@ -1,9 +1,15 @@
-import { type FC, useRef } from 'react';
+import { type ComponentType, useRef } from 'react';
```
Comment (darkskygit) [inline apps/ligo-virgo/src/pages/tools/icons/Icons.tsx:1]: please provide a conversion script - for example, saved in tools/ and replaced by core contributors, too large pr will make it difficult for us to review

Comment (liby) [reply apps/ligo-virgo/src/pages/tools/icons/Icons.tsx:1]: > please provide a conversion script 

> Using [jscodeshift](https://github.com/facebook/jscodeshift) and [codemod-replace-react-fc-typescript](https://github.com/gndelia/codemod-replace-react-fc-typescript)to remove React.FC from components.

Comment (darkskygit) [reply apps/ligo-virgo/src/pages/tools/icons/Icons.tsx:1]: great tools, we've cleaned up the code with it

**Comment by lawvs** [comment]
Has been processed in https://github.com/toeverything/AFFiNE/commit/40b617c42944f1cc12a11482fa2a5b9283cc3fec https://github.com/toeverything/AFFiNE/commit/d3b4906da91da50627a0cf3e5bf5665ff6c2cd61

Thanks for you PR!

**Pull request closed without merging**